src/v0/util/index.js

```diff
@@ -659,6 +659,33 @@ const getValueFromMessage = (message, sourceKeys) => {
   return null;
 };
 
+const getFirstMatchingKeyAndValue = (message, sourceKeys) => {
+  let srcKeys = sourceKeys;
+  if (
+    typeof message !== 'object' ||
+    (!Array.isArray(sourceKeys) && typeof sourceKeys !== 'string')
+  ) {
+    // sourceKeys => boolean, number, object, function etc,.
+    // message => anything other than object,.
+    return { key: '' };
+  }
+  if (typeof sourceKeys === 'string') {
+    srcKeys = [sourceKeys];
+  }
+  // got the possible sourceKeys
+  // eslint-disable-next-line no-restricted-syntax
+  for (const sourceKey of srcKeys) {
+    const val = get(message, sourceKey);
+    if (!lodash.isNil(val)) {
+      // return only if the value is valid.
+      // else look for next possible source in precedence
+      return { value: val, key: sourceKey };
+    }
+  }
+
+  return { value: null, key: '' };
```

Comment:
Missed changing the value to undefined here as well ?

Reply:
Done